### DISCUSSION TITLE
How to define nested fields

### DISCUSSION THREAD

**Discussion by marefati110** [react-querybuilder/react-querybuilder#550] [Q&A]
How can I define nested fields? Imagine database fields are like this:

```js
{
    "name": string,
     items: [
         {name:string}
      ],
      section: {  sectionName: string  }
}
```

It's a common pattern in MongoDB or json/jsonb column type in Postgres.

Does `react-querybuilder` support this or not?

**Comment by jakeboone02** [comment]
My first instinct is to recommend using dot notation in your `field` names, like this:

```ts
const fields: Field[] = [
  { name: "section.sectionName", label: "Section > Section Name" }
];
```

I'm not sure if `formatQuery(query, 'mongodb')` would handle that in a useful way. I may look into that.

If you need a nested/specialized field selector, [`react-select`](https://react-select.com/) is good and the [docs](https://react-querybuilder.js.org/) have some examples that use it.

You also might find the discussion in #327 helpful.
